hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/AmqpMessageDispatcherService.java

```diff
@@ -304,16 +303,15 @@ protected void targetCancelAssignmentToDistributionSet(final CancelTargetAssignm
             return;
         }
 
-        final Optional<Target> eventEntity = cancelEvent.getEntity();
-        if (eventEntity.isPresent()) {
-            final Target target = eventEntity.get();
-            sendCancelMessageToTarget(cancelEvent.getTenant(), target.getControllerId(), cancelEvent.getActionId(),
-                    target.getAddress());
-        } else {
-            LOG.warn(
-                    "Cannot process the received CancelTargetAssignmentEvent with action ID {} because the referenced target with ID {} does no longer exist.",
-                    cancelEvent.getActionId(), cancelEvent.getEntityId());
-        }
+        final List<Target> eventTargets = targetManagement.getByControllerID(cancelEvent.getActions().keySet());
```

Comment:
Please make sure that controller ids parameter size is not larger than 1000 (e.g. use partitioning or validation constraint), also consider `TargetAssignDistributionSetEvent`

Reply:
Introduced logic for partitioned execution considering both kind of events